pkg/snapshot/generate.go

```diff
@@ -0,0 +1,96 @@
+package snapshot
+
+import (
+	"encoding/json"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	liquiditytypes "github.com/gravity-devs/liquidity/v2/x/liquidity/types"
+
+	"github.com/ignite/cli-plugin-airdrop/pkg/encode"
+)
+
+// Generate produce the snapshot of address with the total of atom balance liquid,
+// staked, bounded and unbonding stake
+func Generate(genState map[string]json.RawMessage) (Snapshot, error) {
+	var (
+		marshaller   = encode.Codec()
+		snapshotAccs = make(Accounts)
+	)
+
+	var bankGenesis banktypes.GenesisState
+	if len(genState[banktypes.ModuleName]) > 0 {
+		err := marshaller.UnmarshalJSON(genState[banktypes.ModuleName], &bankGenesis)
+		if err != nil {
+			return Snapshot{}, err
+		}
+	}
+	for _, balance := range bankGenesis.Balances {
+		var (
+			address = balance.Address
+			acc     = snapshotAccs.getAccount(address)
+		)
+		acc.Balance = balance.Coins
+		snapshotAccs[address] = acc
+	}
+
+	var stakingGenesis stakingtypes.GenesisState
+	if len(genState[stakingtypes.ModuleName]) > 0 {
+		err := marshaller.UnmarshalJSON(genState[stakingtypes.ModuleName], &stakingGenesis)
+		if err != nil {
+			return Snapshot{}, err
+		}
+	}
+	for _, unbonding := range stakingGenesis.UnbondingDelegations {
+		var (
+			address        = unbonding.DelegatorAddress
+			acc            = snapshotAccs.getAccount(address)
+			unbondingStake = sdk.NewInt(0)
+		)
+
+		for _, entry := range unbonding.Entries {
+			unbondingStake = unbondingStake.Add(entry.Balance)
+		}
+
+		acc.UnbondingStake = acc.UnbondingStake.Add(unbondingStake)
+		snapshotAccs[address] = acc
+	}
+
+	// Make a map from validator operator address to the v036 validator type
+	validators := make(map[string]stakingtypes.Validator)
+	for _, validator := range stakingGenesis.Validators {
+		validators[validator.OperatorAddress] = validator
+	}
+
+	for _, delegation := range stakingGenesis.Delegations {
+		var (
+			address = delegation.DelegatorAddress
+			acc     = snapshotAccs.getAccount(address)
+			val     = validators[delegation.ValidatorAddress]
+			staked  = delegation.Shares.MulInt(val.Tokens).Quo(val.DelegatorShares).RoundInt()
+		)
+		acc.Staked = acc.Staked.Add(staked)
+		snapshotAccs[address] = acc
+	}
+
+	var liquidityGenesis liquiditytypes.GenesisState
```

Comment:
What is liquidity?
From my understanding, we parse a genesis for Cosmos-hub which doesn't contain liquidity pool since Gravity DEX has been removed
In addition, I think we should focus the first iteration on token delegation to determine airdrop allocation and put liquidity providing for future functionalities